server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -136,14 +136,49 @@ public void onFailure(Exception e) {
      * Creates a listener that wraps another listener, mapping response values via the given mapping function and passing along
      * exceptions to the delegate.
      *
-     * @param listener Listener to delegate to
+     * Notice that it is considered a bug if the listener's onResponse or onFailure fails. onResponse failures will not call onFailure.
+     *
+     * If the function fails, the listener's onFailure handler will be called. The principle is that the code using this is responsible for
```

Comment:
Maybe word this a little differently, this reads somewhat confusing (IMO):

```
The principle is that the mapped listener will handle exceptions from the mapping function {@code fn} but it is the responsibility of {@code delegate} to handle its own exceptions inside `onResponse` and `onFailure`.
```

?

Reply:
Thanks Armin, I used your improved comment here: 09b55769e9e3aab45789b019d362111f51fc8e35